cpp/src/arrow/compute/kernels/CMakeLists.txt

```diff
@@ -72,6 +72,12 @@ add_arrow_compute_test(scalar_utility_test
                        EXTRA_LINK_LIBS
                        arrow_compute_kernels_testing)
 
+add_arrow_compute_test(scalar_string_test
+                       SOURCES
+                       scalar_string_test.cc
```

Comment:
This file is already compiled into `scalar_type_test`  above, so you don't need to add a new test declaration for it.